apps/new-dealer/index.js

```diff
@@ -74,12 +76,40 @@ module.exports = {
       }
     },
     '/supporting-docs': {
-      next: 'weapons',
+      next: '/ammunition',
       forks: [{
-        target: '/ammunition',
-        condition: ammunition
+        target: '/storage-weapons-ammo',
+        condition(req) {
+          const stored = storedOnPremises(req, 'true');
+          const weaponsAndAmmo = weapons(req) && ammunition(req);
+          return stored && weaponsAndAmmo;
+        }
+      }, {
+        target: '/storage-address',
+        condition(req) {
+          const stored = storedOnPremises(req, 'true');
+          const weaponsAndAmmo = weapons(req) && ammunition(req);
+          return stored && !weaponsAndAmmo;
+        }
+      }, {
+        target: '/weapons',
+        condition(req) {
+          const stored = storedOnPremises(req, 'false');
```

Comment:
hey Friend, I think this is a confusing use case for this function as it acts differently in different cases. It's like you're telling it what to expect.

Could you refactor the storedOnPremises function to remove the second arg?
```js
function storedOnPremesis(req) {
  return req.sessionModel.get('stored-on-premesis') === 'true';
}
```